### PR TITLE
[Remove Stream/Optional: Phase 3] LPS-181532 module: headless-delivery-Impl

### PR DESCRIPTION
**Pull request by dantewang**

### CODE REVIEW

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/NavigationMenuResourceImpl.java [hunk 1]
```diff
@@ -713,8 +713,7 @@ private void _updateNavigationMenuItems(
 						siteNavigationMenuItem.getSiteNavigationMenuItemId(),
 						siteId, siteNavigationMenuId);
 
-					siteNavigationMenuItems.remove(
-						existingSiteNavigationMenuItem);
+					siteNavigationMenuItems.remove(siteNavigationMenuItem);
```
Comment (JanisZhang) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/NavigationMenuResourceImpl.java]: Hi @dantewang, before SF, here the logic was to delete the original object, now it seems to remove the updated object. I'm not sure whether it's ok or not.

**Comment by dantewang** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
ci:test:headless

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/NavigationMenuResourceImpl.java (hunk 1 above)
Comment (dantewang) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/NavigationMenuResourceImpl.java:717]: @JanisZhang Yeah, it should be another object. Updated, thanks!

Comment (JanisZhang) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/NavigationMenuResourceImpl.java:717]: Thank you!

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by dantewang** [comment]
ci:forward

cc @shuyangzhou @tinatian

**Pull request closed without merging**